package de.paula.projectmanagement.project;

import jakarta.persistence.Embeddable;

import java.io.Serializable;
import java.util.UUID;
import java.util.Objects;

@Embeddable
public class ProjectMemberId implements Serializable {

  private UUID projectId;
  private UUID userId;

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }

    if (!(o instanceof ProjectMemberId that)) {
      return false;
    }

    return Objects.equals(projectId, that.projectId)
            && Objects.equals(userId, that.userId);
  }


  @Override
  public int hashCode() {
    return Objects.hash(projectId, userId);
  }

}